File: abc_C/255_+-1_operation.cpp
#include <bits/stdc++.h>
using ll = long long;
using namespace std;

int main(void) {
  ll X, A, D, N;
  cin >> X >> A >> D >> N;

  ll ans;
  if (D == 0) {
    ans = abs(X - A);
  } else {
    ll K = (X - A) / D;
    if (K < 0) {
      ans = abs(X - A);
    } else if (K > N - 1) {
      ans = abs(X - A - (N-1)*D);
    } else {
      ans = 1000000000;
      for(ll i=-2; i<=2; i++) {
        ans = min(ans, abs(X - A - (K+i)*D));
      }
    }
  }
  printf("%lld\n", ans);
  // cout << ans << endl;
}

File: ABC_C_/255_+-1_operation.py
import numpy as np

X, A, D, N = map(int, input().split())

if D == 0:
  ans = abs(X-A)
else:
  k = (X-A) // D
  if k < 0:
    k = 0
    ans = abs(X-A-k*D)
  elif k > N-1:
    k = N-1
    ans = abs(X-A-k*D)
  else:
    ans = abs(X-A-k*D)
    for i in range(-2, 3):
      ans = min( ans, abs(X-A-(k+i)*D) )

print(ans)